uber/templates/macros.html

```diff
@@ -21,46 +21,205 @@
 {%- endmacro %}
 
 
-{% macro checkbox(model, field_name, id=None, label=None, label_class=None, readonly=False, title=None) -%}
+{% macro checkbox(model, field_name, id=None, label=None, label_class=None, is_readonly=False, title=None) -%}
     {% if label %}
       <label
-        class="checkbox-label{% if label_class %} {{ label_class }}{% endif %}"
-        {% if title %}title="{{ title }}"{% endif %}>
+          class="checkbox-label{% if label_class %} {{ label_class }}{% endif %}{% if is_readonly %} disabled{% endif %}"
+          {% if title %}title="{{ title }}"{% endif %}>
     {% endif %}
-    {% if readonly %}
+    {% if is_readonly %}
       <input
-        type="hidden"
-        name="{{ field_name }}"
-        id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
-        value="{%- if model[field_name] -%}1{%- else -%}0{%- endif -%}" />
+          type="hidden"
+          name="{{ field_name }}"
+          id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
+          value="{%- if model[field_name] -%}1{%- else -%}0{%- endif -%}" />
       <input
-        type="checkbox"
-        value="1"
-        disabled
-        {% if not label and title %}title="{{ title }}"{% endif %}
-        {% if model[field_name] %}checked{% endif %} />
+          type="checkbox"
+          value="1"
+          disabled
+          {% if not label and title %}title="{{ title }}"{% endif %}
+          {% if model[field_name] %}checked{% endif %} />
     {% else %}
       <input
-        type="checkbox"
-        name="{{ field_name }}"
-        id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
-        value="1"
-        {% if not label and title %}title="{{ title }}"{% endif %}
-        {% if model[field_name] %}checked{% endif %} />
+          type="checkbox"
+          name="{{ field_name }}"
+          id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
+          value="1"
+          {% if not label and title %}title="{{ title }}"{% endif %}
+          {% if model[field_name] %}checked{% endif %} />
     {% endif %}
     {% if label %}{{ label }}</label>{% endif %}
 {%- endmacro %}
 
 
-{% macro checkgroup(model, field_name, defaults=False, no_instance=False) -%}
-    {%- set defaults = model[field_name] if not defaults and not no_instance else None -%}
-    {%- set defaults = defaults.split(",") if defaults else [] -%}
+{% macro checkgroup(model, field_name, suffix='', defaults=False, is_readonly=False) -%}
+  {%- set is_class = model is class -%}
+  {%- set defaults = model[field_name] if not defaults and not is_class else '' -%}
```

Comment:
Wait, doesn't this line mean that it will override defaults passed in as long as `is_class` is true? Is that going to be a problem anywhere? I don't currently understand what `is_class` is supposed to actually mean.